File: 백준/Silver/2559. 수열/수열.cc
#include <bits/stdc++.h>
using namespace std;

const int INF = 1e9;
int n, k, temp, ret = -INF, psum[100004];

int main() {
  ios_base::sync_with_stdio(false);
  cin.tie(NULL);
  cout.tie(NULL);

  cin >> n >> k;
  for (int i = 1; i <= n; i++) {
    cin >> temp;
    psum[i] = psum[i - 1] + temp;
  }

  for (int i = k; i <= n; i++) {
    int sum = psum[i] - psum[i - k];

    ret = max(ret, sum);
  }

  cout << ret;

  return 0;
}

File: 백준/Silver/2559. 수열/수열.js
const fs = require("fs");
const input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');

const [n, k] = input[0].split(' ').map(a => Number(a));
const temps = input[1].split(' ').map(a => Number(a));

function answer(n, k, temps) {
    const psum = [0];
    let ret = -Infinity;

    for (let i = 1; i <= n; i++){
        psum[i] = psum[i - 1] + temps[i - 1];
    }
    
    for (let i = k; i <= n; i++){
        let sum = psum[i] - psum[i - k];

        ret = Math.max(sum, ret);
    }
    
    console.log(ret);
}

answer(n, k, temps);

